tool/gravity/cli/operation.go

```diff
@@ -197,59 +206,59 @@ func completeClusterOperationPlan(localEnv *localenv.LocalEnvironment, operation
+// Returns nil if there is a completed install operation
+func CheckInstallOperationComplete(localEnv *localenv.LocalEnvironment) error {
+	operations, err := getBackendOperations(localEnv, nil, "")
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return trace.Wrap(err)
 	}
-	log.WithField("operations", oplist(operations).String()).Debug("Fetched backend operations.")
+	log.WithField("operations", operationList(operations).String()).Debug("Fetched backend operations.")
 	if len(operations) == 0 {
-		if operationID != "" {
-			return nil, trace.NotFound("no operation with ID %v found", operationID)
-		}
-		return nil, trace.NotFound("no operation found")
-	}
-	if len(operations) == 1 && operationID != "" {
-		log.WithField("operation", operations[0]).Debug("Fetched operation by ID.")
-		return &operations[0], nil
+		return trace.NotFound("no install operation found")
 	}
-	if len(operations) != 1 {
-		log.Infof("Multiple operations found: \n%v\n, please specify operation with --operation-id.\n"+
-			"Displaying the most recent operation.",
-			oplist(operations))
+	firstOperation := operations[len(operations)-1]
+	if firstOperation.Type == ops.OperationInstall && firstOperation.IsCompleted() {
+		return nil
 	}
-	return &operations[0], nil
+	return trace.NotFound("no install operation found")
 }
 
-func getActiveOperation(localEnv *localenv.LocalEnvironment, environ LocalEnvironmentFactory, operationID string) (*ops.SiteOperation, error) {
+// getLastOperation returns the last operation found across the specified backends.
+// If no operation is found, the returned error will indicate a not found operation
+func getLastOperation(localEnv *localenv.LocalEnvironment, environ LocalEnvironmentFactory, operationID string) (*clusterOperation, error) {
 	operations, err := getBackendOperations(localEnv, environ, operationID)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	log.WithField("operations", oplist(operations).String()).Debug("Fetched backend operations.")
+	log.WithField("operations", operationList(operations).String()).Debug("Fetched backend operations.")
 	if len(operations) == 0 {
 		if operationID != "" {
-			return nil, trace.NotFound("no operation with ID %v found", operationID)
+			return nil, newOperationNotFound("no operation with ID %v found", operationID)
 		}
-		return nil, trace.NotFound("no operation found")
+		return nil, newOperationNotFound("no operation found")
 	}
-	op, err := getActiveOperationFromList(operations)
+	return &operations[0], nil
+}
+
+func getActiveOperation(localEnv *localenv.LocalEnvironment, environ LocalEnvironmentFactory, operationID string) (*clusterOperation, error) {
+	operation, err := getLastOperation(localEnv, environ, operationID)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	return op, nil
+	if operation.IsCompleted() {
+		return nil, operationNotFound{message: "no active operation found", existing: true}
```

Comment:
With this change, will we still be able to see a plan for a completed operation?

Reply:
When displaying the plan, this function is not used - `getLastOperation` is used instead.